### PR TITLE
chore(docs): migrate datepicker docs to storybook

### PR DESCRIPTION
**Pull request by austingreendev**
## Description

This PR migrates our `Datepicker` examples to Storybook.

## Detail

During this migration, I noticed an issue with our DatepickerRange component not disabling hover styling when outside of the `min/max` bounds. I've included a fix for the issue in this PR.

## Checklist

- [ ] ~:ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] ~:guardsman: includes new unit tests~
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/datepickers/stories/4-Range.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import React, { useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { Meta, Story } from '@storybook/react';
+import { DatepickerRange } from '@zendeskgarden/react-datepickers';
+import { Grid, Row, Col } from '@zendeskgarden/react-grid';
+import { Field, Label, Input, Message } from '@zendeskgarden/react-forms';
+import { addDays, compareAsc } from 'date-fns';
+
+export default {
+  title: 'Components/Datepicker/DatepickerRange',
```
Comment (jzempel) [inline packages/datepickers/stories/4-Range.stories.tsx:17]: Can we get all of these tucked under a single `Datepickers` folder?

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/datepickers/stories/4-Range.stories.tsx (hunk 1 above)
Comment (austingreendev) [reply packages/datepickers/stories/4-Range.stories.tsx:17]: This was unintentionally caused by a rebase. Everything is now under `Components/Datepickers`

File: demo/index.html [hunk 2]
```diff
@@ -110,7 +110,7 @@ <h1 class="c-main__title">React Components</h1>
               <a class="u-fg-inherit" href="storybook/?path=/story/components-chrome-readme--page">Chrome</a>
             </div>
             <div class="u-mb-sm">
-              <a class="u-fg-inherit" href="datepickers">Datepickers</a>
+              <a class="u-fg-inherit" href="storybook/?path=/story/components-datepicker-readme--page">Datepickers</a>
```
[the remark this answers is not in the document]
Comment (austingreendev) [reply demo/index.html:113]: This was caused by a bad rebase. This now references the root `Datepickers` folder.

File: demo/index.html [hunk 3]
```diff
@@ -110,8 +110,8 @@ <h1 class="c-main__title">React Components</h1>
               <a class="u-fg-inherit" href="storybook/?path=/story/components-chrome-readme--page">Chrome</a>
             </div>
             <div class="u-mb-sm">
-              <a class="u-fg-inherit" href="datepickers">Datepickers</a>
-            </div>
+              <a class="u-fg-inherit" href="storybook/?path=/story/components-datepickers-readme--page">Datepickers</a>
+            </div>e
```
Comment (jzempel) [inline demo/index.html:114]: remove stray "e"

**Review by jzempel — CHANGES_REQUESTED** [review]

**Review by jzempel — APPROVED** [review]

**Pull request merged**